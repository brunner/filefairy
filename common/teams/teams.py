#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""Common (non-reloadable) util methods for team information."""

import re
from functools import partial


def _team(encoding, abbreviation, hometown, nickname):
    special1 = encoding in ['T30', 'TCH', 'TLA', 'TNY']
    special2 = encoding in ['T30', 'T35', 'T36', 'T44', 'T45', 'T48', 'T49']

    if encoding == 'TCH':
        encodings = ['T35', 'T36']
    elif encoding == 'TLA':
        encodings = ['T44', 'T45']
    elif encoding == 'TNY':
        encodings = ['T48', 'T49']
    else:
        encodings = [encoding]

    return {
        'abbreviation': abbreviation if not special1 else '',
        'decoding': (hometown + ' ' + nickname) if not special1 else '',
        'encoding': encoding,
        'encodings': encodings,
        'hometown': hometown,
        'lower': nickname.replace(' ', '').lower(),
        'nickname': nickname,
        'precoding': hometown if not special2 else '',
        'teamid': encoding.strip('T') if not special1 else '',
    }


TEAMS = [
    _team('T30', 'OBL', 'OBL', 'OBL'),
    _team('T31', 'ARI', 'Arizona', 'Diamondbacks'),
    _team('T32', 'ATL', 'Atlanta', 'Braves'),
    _team('T33', 'BAL', 'Baltimore', 'Orioles'),
    _team('T34', 'BOS', 'Boston', 'Red Sox'),
    _team('T35', 'CWS', 'Chicago', 'White Sox'),
    _team('T36', 'CHC', 'Chicago', 'Cubs'),
    _team('T37', 'CIN', 'Cincinnati', 'Reds'),
    _team('T38', 'CLE', 'Cleveland', 'Indians'),
    _team('T39', 'COL', 'Colorado', 'Rockies'),
    _team('T40', 'DET', 'Detroit', 'Tigers'),
    _team('T41', 'MIA', 'Miami', 'Marlins'),
    _team('T42', 'HOU', 'Houston', 'Astros'),
    _team('T43', 'KC', 'Kansas City', 'Royals'),
    _team('T44', 'LAA', 'Los Angeles', 'Angels'),
    _team('T45', 'LAD', 'Los Angeles', 'Dodgers'),
    _team('T46', 'MIL', 'Milwaukee', 'Brewers'),
    _team('T47', 'MIN', 'Minnesota', 'Twins'),
    _team('T48', 'NYY', 'New York', 'Yankees'),
    _team('T49', 'NYM', 'New York', 'Mets'),
    _team('T50', 'OAK', 'Oakland', 'Athletics'),
    _team('T51', 'PHI', 'Philadelphia', 'Phillies'),
    _team('T52', 'PIT', 'Pittsburgh', 'Pirates'),
    _team('T53', 'SD', 'San Diego', 'Padres'),
    _team('T54', 'SEA', 'Seattle', 'Mariners'),
    _team('T55', 'SF', 'San Francisco', 'Giants'),
    _team('T56', 'STL', 'St. Louis', 'Cardinals'),
    _team('T57', 'TB', 'Tampa Bay', 'Rays'),
    _team('T58', 'TEX', 'Texas', 'Rangers'),
    _team('T59', 'TOR', 'Toronto', 'Blue Jays'),
    _team('T60', 'WAS', 'Washington', 'Nationals'),
    _team('TCH', '', 'Chicago', ''),
    _team('TNY', '', 'New York', ''),
    _team('TLA', '', 'Los Angeles', ''),
]


def _map(k, vs):
    return {t[k]: {v: t[v] for v in vs if t[v]} for t in TEAMS if t[k]}


def _repl(f, m):
    a = m.group(0)
    b = f(a)
    return b if b else a


def _sub(ks, f, s):
    pattern = '|'.join(ks)
    return re.sub(pattern, partial(_repl, f), s)


DECODING_MAP = _map('decoding', ['encoding'])

ENCODING_MAP = _map('encoding', [
    'abbreviation',
    'decoding',
    'encodings',
    'hometown',
    'lower',
    'nickname',
    'teamid',
])

PRECODING_MAP = _map('precoding', ['encoding'])

ICON_LINK = 'https://fairylab.surge.sh/images/teams/{0}/{0}-icon.png'
MODAL_LINK = ' data-toggle="modal" data-target="#{}"'
BADGE_TAG = '<span class="badge badge-icon badge-light"{}>{}</span>'
IMG_TAG = '<img src="{}" width="16" height="16" border="0" class="{}">'
SPAN_TAG = '<span class="{}">{}</span>'


def decoding_to_encoding(decoding):
    return DECODING_MAP.get(decoding, {}).get('encoding', '')


def decoding_to_encoding_sub(text):
    return _sub(DECODING_KEYS, decoding_to_encoding, text)


def encoding_keys():
    return ENCODING_KEYS


def encoding_to_abbreviation(encoding):
    return ENCODING_MAP.get(encoding, {}).get('abbreviation', '')


def encoding_to_colors(encoding):
    return ENCODING_MAP.get(encoding, {}).get('colors', [])


def encoding_to_decoding(encoding):
    return ENCODING_MAP.get(encoding, {}).get('decoding', '')


def encoding_to_decoding_sub(text):
    return _sub(ENCODING_KEYS, encoding_to_decoding, text)


def encoding_to_encodings(encoding):
    return ENCODING_MAP.get(encoding, {}).get('encodings', '')


def encoding_to_hometown(encoding):
    return ENCODING_MAP.get(encoding, {}).get('hometown', '')


def encoding_to_hometown_sub(text):
    return _sub(ENCODING_KEYS, encoding_to_hometown, text)


def encoding_to_lower(encoding):
    return ENCODING_MAP.get(encoding, {}).get('lower', '')


def encoding_to_nickname(encoding):
    return ENCODING_MAP.get(encoding, {}).get('nickname', '')


def encoding_to_teamid(encoding):
    return ENCODING_MAP.get(encoding, {}).get('teamid', '')


def icon_absolute(encoding, text):
    lower = encoding_to_lower(encoding)
    ic = 'absolute-icon-image'
    sc = 'd-block pl-4'

    img = IMG_TAG.format(ICON_LINK.format(lower), ic)
    span = SPAN_TAG.format(sc, text)
    return img + span


def icon_badge(encoding, text, active):
    lower = encoding_to_lower(encoding)
    teamid = encoding_to_teamid(encoding)
    ic = 'badge-icon-image'
    sc = 'badge-icon-text align-middle'

    if active:
        ba = MODAL_LINK.format(teamid)
        if text == '0-0':
            text = '?-?'
    else:
        ba = ''
        ic += ' grayscale'
        sc += ' text-secondary'

    img = IMG_TAG.format(ICON_LINK.format(lower), ic)
    span = SPAN_TAG.format(sc, text.replace('-', ' - '))
    return BADGE_TAG.format(ba, img + span)


def precoding_to_encoding(precoding):
    return PRECODING_MAP.get(precoding, {}).get('encoding', '')


def precoding_to_encoding_sub(text):
    return _sub(PRECODING_KEYS, precoding_to_encoding, text)


DECODING_KEYS = list(sorted(DECODING_MAP.keys()))
ENCODING_KEYS = list(sorted(ENCODING_MAP.keys()))
PRECODING_KEYS = list(sorted(PRECODING_MAP.keys()))
