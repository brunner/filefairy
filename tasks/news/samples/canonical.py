#!/usr/bin/env python3
# -*- coding: utf-8 -*-

import os
import re
import sys

_path = os.path.dirname(os.path.abspath(__file__))
sys.path.append(re.sub(r'/tasks/news/samples', '', _path))

from common.elements.elements import cell  # noqa
from common.elements.elements import col  # noqa
from common.elements.elements import row  # noqa
from common.elements.elements import table  # noqa
from common.teams.teams import icon_absolute  # noqa

_injuries = [
    table(
        clazz='border mb-3',
        hcols=[col(clazz='font-weight-bold text-dark')],
        bcols=[col(clazz='position-relative')],
        head=[row(cells=[cell(content='Monday, July 4th, 2022')])],
        body=[
            row(cells=[
                cell(
                    content=icon_absolute('T52', (
                        'SP <a href="https://statsplus.net/oblootp/player/3515'
                        '5">John Barbato</a> diagnosed with a sore forearm, is'
                        ' day-to-day for 5 days.')))
            ]),
        ]),
    table(
        clazz='border mb-3',
        hcols=[col(clazz='font-weight-bold text-dark')],
        bcols=[col(clazz='position-relative')],
        head=[row(cells=[cell(content='Sunday, July 3th, 2022')])],
        body=[
            row(cells=[
                cell(
                    content=icon_absolute('T44', (
                        'SP <a href="https://statsplus.net/oblootp/player/3731'
                        '8">Ty Blach</a> was injured while pitching. Diagnosis'
                        ': back stiffness. This is a day-to-day injury expecte'
                        'd to last 5 days.')))
            ]),
            row(cells=[
                cell(
                    content=icon_absolute('T60', (
                        'RP <a href="https://statsplus.net/oblootp/player/3262'
                        '4">Hunter Strickland</a> was injured while pitching. '
                        'Diagnosis: back spasms. This is a day-to-day injury e'
                        'xpected to last 5 days.')))
            ]),
            row(cells=[
                cell(
                    content=icon_absolute('T57', (
                        'RP <a href="https://statsplus.net/oblootp/player/1958'
                        '">Javier Medina</a> was injured while pitching. The d'
                        'iagnosis is not yet known.')))
            ]),
            row(cells=[
                cell(
                    content=icon_absolute('T52', (
                        'SP <a href="https://statsplus.net/oblootp/player/3515'
                        '5">John Barbato</a> was injured while pitching. The d'
                        'iagnosis is not yet known.')))
            ]),
            row(cells=[
                cell(
                    content=icon_absolute('T52', (
                        'CF <a href="https://statsplus.net/oblootp/player/1604'
                        '">Carl Cester</a> was injured on a defensive play. Th'
                        'e diagnosis is not yet known.')))
            ]),
            row(cells=[
                cell(
                    content=icon_absolute('T40', (
                        'RP <a href="https://statsplus.net/oblootp/player/2245'
                        '7">Walker Buehler</a> was injured while pitching. The'
                        ' diagnosis is not yet known.')))
            ]),
        ])
]
_news = [
    table(
        clazz='border mb-3',
        hcols=[col(clazz='font-weight-bold text-dark')],
        bcols=[col(clazz='position-relative')],
        head=[row(cells=[cell(content='Monday, July 4th, 2022')])],
        body=[
            row(cells=[
                cell(
                    content=icon_absolute('T35', (
                        '1B <a href="https://statsplus.net/oblootp/player/3389'
                        '2">Viosergy Rosa</a> wins the MLB AL Player of the We'
                        'ek Award.')))
            ]),
            row(cells=[
                cell(
                    content=icon_absolute('T32', (
                        '1B <a href="https://statsplus.net/oblootp/player/2719'
                        '4">Freddie Freeman</a> wins the MLB NL Player of the '
                        'Week Award.')))
            ]),
        ])
]
_transactions = [
    table(
        clazz='border mb-3',
        hcols=[col(clazz='font-weight-bold text-dark')],
        bcols=[col(clazz='position-relative')],
        head=[row(cells=[cell(content='Monday, July 4th, 2022')])],
        body=[
            row(cells=[
                cell(
                    content=icon_absolute('T53', (
                        'Signed <a href="https://statsplus.net/oblootp/player/'
                        '56862">Antonio Marquis</a> to a minor league contract'
                        ' with a signing bonus of $180,000.')))
            ]),
            row(cells=[
                cell(
                    content=icon_absolute('T30', (
                        'The Detroit Tigers traded 25-year-old minor league le'
                        'ft fielder <a href="https://statsplus.net/oblootp/pla'
                        'yer/28672">Nick Plummer</a> to the Milwaukee Brewers,'
                        ' getting 33-year-old closer <a href="https://statsplu'
                        's.net/oblootp/player/36352">Phil Klein</a> in return.'
                    )))
            ]),
            row(cells=[
                cell(
                    content=icon_absolute('T30', (
                        'The Detroit Tigers traded 25-year-old minor league le'
                        'ft fielder <a href="https://statsplus.net/oblootp/pla'
                        'yer/25202">Wesley Rodriguez</a> and 31-year-old left '
                        'fielder <a href="https://statsplus.net/oblootp/player'
                        '/33588">Michael A. Taylor</a> to the Chicago Cubs, ge'
                        'tting 30-year-old right fielder <a href="https://stat'
                        'splus.net/oblootp/player/35161">Drew Vettleson</a> in'
                        ' return.')))
            ]),
            row(cells=[
                cell(
                    content=icon_absolute('T30', (
                        'The Detroit Tigers traded 21-year-old minor league sh'
                        'ortstop <a href="https://statsplus.net/oblootp/player'
                        '/32327">Jesus Martinez</a> to the Philadelphia Philli'
                        'es, getting 28-year-old left fielder <a href="https:/'
                        '/statsplus.net/oblootp/player/37237">Mitch Nay</a> in'
                        ' return.')))
            ]),
            row(cells=[
                cell(
                    content=icon_absolute('T30', (
                        'The Minnesota Twins traded 25-year-old first baseman '
                        '<a href="https://statsplus.net/oblootp/player/39581">'
                        'Lewin Diaz</a> to the Chicago Cubs, getting 30-year-o'
                        'ld starting pitcher <a href="https://statsplus.net/ob'
                        'lootp/player/37975">Matt Anderson</a>, 19-year-old mi'
                        'nor league starting pitcher <a href="https://statsplu'
                        's.net/oblootp/player/22921">Juan Monreal</a> and 21-y'
                        'ear-old minor league catcher <a href="https://statspl'
                        'us.net/oblootp/player/55093">Jonathan Salazar</a> in '
                        'return.')))
            ]),
        ])
]

subtitle = ''

tmpl = 'news.html'

context = {'injuries': _injuries, 'news': _news, 'transactions': _transactions}
