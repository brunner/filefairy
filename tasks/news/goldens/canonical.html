<!doctype html>
<html lang="en">

<head>

<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link rel="apple-touch-icon" sizes="120x120" href="/apple-touch-icon.png">
<link rel="icon" type="image/png" sizes="32x32" href="/favicon-32x32.png">
<link rel="icon" type="image/png" sizes="16x16" href="/favicon-16x16.png">
<link rel="manifest" href="/site.webmanifest">
<meta name="msapplication-TileColor" content="#dfe3e7">
<meta name="theme-color" content="#dfe3e7">
<link rel="stylesheet"
      href="https://maxcdn.bootstrapcdn.com/bootstrap/4.0.0/css/bootstrap.min.css"
      integrity="sha384-Gn5384xqQ1aoWXA+058RXPxPg6fy4IWvTNh0E263XmFcJlSAwiGgFAW/dAiS6JXm"
      crossorigin="anonymous">
<link rel="stylesheet"
      href="https://cdnjs.cloudflare.com/ajax/libs/open-iconic/1.1.1/font/css/open-iconic-bootstrap.min.css">
<title>Fairylab Â· News</title>

<style>
html {
  min-height: 100%;
  overflow-y: scroll;
  position: relative;
}
header {
  background-color: #f5f5f5;
  width: 100%;
}
pre {
  margin: 0;
  white-space: pre-wrap;
  white-space: -moz-pre-wrap;
  white-space: -pre-wrap;
  white-space: -o-pre-wrap;
  word-wrap: break-word;
}
a.disabled{
  pointer-events: none;
  cursor: default;
}
button.close {
  height: 16px;
  line-height: 16px;
  width: 16px;
}
span.absolute-icon {
  border-radius: .25rem;
  padding: 3px;
  top: 12px;
}
.absolute-icon {
  position: absolute;
  top: 13px;
}
.absolute-icon.left {
  left: 10px;
}
.absolute-icon.right {
  left: auto;
  right: 10px;
}
.absolute-icon.hover:hover {
  background-color: #e9ecef;
  cursor: pointer;
}
.head {
  border-left: 1px solid transparent;
  border-right: 1px solid transparent;
  padding: .75rem;
}
.table {
  margin-bottom: 0;
  word-break: break-word;
}
.table .thead-light td {
  background-color: #f5f5f5;
}
.table-fixed {
  table-layout: fixed;
}
.small.table td {
  padding: .75rem .5rem;
}
.small.table td:first-child {
  padding-left: .75rem;
}
.small.table td:last-child {
  padding-right: .75rem;
}
.small.table td.badge-icon-wrapper {
  padding: .5rem .25rem;
}
.small.topper {
  font-size: 90%;
}
.table.collapse.show {
  display: table;
}
.td-sm-none,
.td-md-none,
.td-lg-none {
  display: none;
}
.td-md-show {
  display: table-cell;
}
@media (min-width: 576px) {
  .td-sm-none {
    display: table-cell;
  }
}
@media (min-width: 768px) {
  .td-md-none {
    display: table-cell;
  }
  .td-md-show {
    display: none
  }
}
@media (min-width: 992px) {
  .td-lg-none {
    display: table-cell;
  }
}
.text-underline {
  text-decoration: underline;
}
.tag {
  background-color: #ffffff;
  line-height: 16px;
  padding: 0 .75em;
  vertical-align: top;
}
.tag-light {
  border-color: #1b1e21!important;
  color: #1b1e21;
}

.carousel-fade .carousel-item {
  opacity: 0;
  transition-duration: .6s;
  transition-property: opacity;
}
.carousel-fade .carousel-item.active,
.carousel-fade .carousel-item-next.carousel-item-left,
.carousel-fade .carousel-item-prev.carousel-item-right {
  opacity: 1;
}
.carousel-fade .active.carousel-item-left,
.carousel-fade .active.carousel-item-right {
  opacity: 0;
}
.carousel-fade .carousel-item-next,
.carousel-fade .carousel-item-prev,
.carousel-fade .carousel-item.active,
.carousel-fade .active.carousel-item-left,
.carousel-fade .active.carousel-item-prev {
  transform: translateX(0);
  transform: translate3d(0, 0, 0);
}

.modal {
  overflow-y: auto;
}
.modal-header {
  background-color: #f5f5f5;
  border-bottom: 1px solid #dee2e6;
}
.modal-open {
  overflow: auto;
  padding-right: 0px !important;
}
@media (min-width: 576px) {
  body.modal-open header {
    margin-right: 0!important;
    padding-right: 0!important;
  }
}
@media (min-width: 768px) {
  .modal-dialog {
    max-width: 650px; 
  }
}
@media (min-width: 992px) {
  .modal-dialog {
    max-width: 800px; 
  }
}

.pagination {
  margin: 0;
  position: absolute;
  right: 0;
  top: 4px;
}
.page-item.active .page-link {
  background-color: #e9ecef;
  border-color: #dee2e6;
}
.page-link {
  background-color: #f8f9fa;
  line-height: 1;
  padding: .25rem .75rem;
}
.page-link-inner {
  color: #212529;
  display: inline-block;
  padding: 0;
  padding-top: 1px;
}

.rt-10p {
  right: 10px;
  top: 10px;
}
.right-8p {
  right: 8px;
}
.top-0 {
  top: 0;
}

.h-58p {
  height: 58px;
}
.h-72p {
  height: 72px;
}
.mr-8p {
  margin-right: 8px;
}
.pr-75r {
  padding-right: .75rem;
}
.w-10 {
  width: 10%;
}
.w-20 {
  width: 20%;
}
.w-40 {
  width: 40%;
}
.w-50 {
  width: 50%;
}
.w-24p {
  width: 24px;
}
.w-28p {
  width: 28px;
}
.w-32p {
  width: 32px;
}
.w-40p {
  width: 40px;
}
.w-50p {
  width: 50px;
}
.w-54p {
  width: 54px;
}
.w-65p {
  width: 65px;
}
.w-70p {
  width: 70px;
}
.w-75p {
  width: 75px;
}
.w-95p {
  width: 95px;
}
.w-105p {
  width: 105px;
}
.w-150p {
  width: 150px;
}
</style>
</head>

<body>

<!-- ^(base.header) -->
<header class="border-bottom sticky-top">
<div class="container">
<div class="head position-relative small text-secondary">
<img src="https://brunnerj.com/fairylab/favicon-32x32.png"
     width="16" height="16" border="0" class="absolute-icon left">
<div class="px-4">News</div>
<span class="oi oi-menu absolute-icon hover right"
      data-toggle="modal" data-target="#menu"></span></div>
</div>
</header>
<!-- $(base.header) -->

<!-- ^(base.main) -->
<main role="main" class="pt-3">

<!-- ^(news.injuries) -->
<div class="container">

<!-- ^(common.topper) -->
<div class="row">
<div class="col-12">

<!-- ^(common.topper.table) -->
<table class="small table topper">

<!-- ^(common.topper.body) -->
<tbody>
<tr>
<td scope="row" class="border-0 font-weight-bold px-0 text-secondary">Injury Log</td>
</tr>
</tbody>
<!-- $(common.topper.body) -->

</table>
<!-- $(common.topper.table) -->

</div>
</div>
<!-- $(common.topper) -->

<!-- ^(news.row) -->
<div class="row">

<!-- ^(news.column) -->
<div class="col-12">

<!-- ^(common.table) -->
<table class="small table border mb-3">

<!-- ^(common.table.head) -->
<thead class="thead-light">
<tr>
<td scope="col" class="font-weight-bold text-dark">Monday, July 4th, 2022</td>
</tr>
</thead>
<!-- $(common.table.head) -->

<!-- ^(common.table.body) -->
<tbody>
<tr>
<td scope="row" class="position-relative"><img src="https://brunnerj.com/fairylab/images/teams/pirates/pirates-icon.png" width="16" height="16" border="0" class="absolute-icon left"><span class="d-block pl-4">SP <a href="https://statsplus.net/oblootp/player/35155">John Barbato</a> diagnosed with a sore forearm, is day-to-day for 5 days.</span></td>
</tr>
</tbody>
<!-- $(common.table.body) -->

</table>
<!-- $(common.table) -->

<!-- ^(common.table) -->
<table class="small table border mb-3">

<!-- ^(common.table.head) -->
<thead class="thead-light">
<tr>
<td scope="col" class="font-weight-bold text-dark">Sunday, July 3th, 2022</td>
</tr>
</thead>
<!-- $(common.table.head) -->

<!-- ^(common.table.body) -->
<tbody>
<tr>
<td scope="row" class="position-relative"><img src="https://brunnerj.com/fairylab/images/teams/angels/angels-icon.png" width="16" height="16" border="0" class="absolute-icon left"><span class="d-block pl-4">SP <a href="https://statsplus.net/oblootp/player/37318">Ty Blach</a> was injured while pitching. Diagnosis: back stiffness. This is a day-to-day injury expected to last 5 days.</span></td>
</tr>
<tr>
<td scope="row" class="position-relative"><img src="https://brunnerj.com/fairylab/images/teams/nationals/nationals-icon.png" width="16" height="16" border="0" class="absolute-icon left"><span class="d-block pl-4">RP <a href="https://statsplus.net/oblootp/player/32624">Hunter Strickland</a> was injured while pitching. Diagnosis: back spasms. This is a day-to-day injury expected to last 5 days.</span></td>
</tr>
<tr>
<td scope="row" class="position-relative"><img src="https://brunnerj.com/fairylab/images/teams/rays/rays-icon.png" width="16" height="16" border="0" class="absolute-icon left"><span class="d-block pl-4">RP <a href="https://statsplus.net/oblootp/player/1958">Javier Medina</a> was injured while pitching. The diagnosis is not yet known.</span></td>
</tr>
<tr>
<td scope="row" class="position-relative"><img src="https://brunnerj.com/fairylab/images/teams/pirates/pirates-icon.png" width="16" height="16" border="0" class="absolute-icon left"><span class="d-block pl-4">SP <a href="https://statsplus.net/oblootp/player/35155">John Barbato</a> was injured while pitching. The diagnosis is not yet known.</span></td>
</tr>
<tr>
<td scope="row" class="position-relative"><img src="https://brunnerj.com/fairylab/images/teams/pirates/pirates-icon.png" width="16" height="16" border="0" class="absolute-icon left"><span class="d-block pl-4">CF <a href="https://statsplus.net/oblootp/player/1604">Carl Cester</a> was injured on a defensive play. The diagnosis is not yet known.</span></td>
</tr>
<tr>
<td scope="row" class="position-relative"><img src="https://brunnerj.com/fairylab/images/teams/tigers/tigers-icon.png" width="16" height="16" border="0" class="absolute-icon left"><span class="d-block pl-4">RP <a href="https://statsplus.net/oblootp/player/22457">Walker Buehler</a> was injured while pitching. The diagnosis is not yet known.</span></td>
</tr>
</tbody>
<!-- $(common.table.body) -->

</table>
<!-- $(common.table) -->

</div>
<!-- $(news.column) -->

</div>
<!-- $(news.row) -->

</div>
<!-- $(news.injuries) -->

<!-- ^(news.news) -->
<div class="container">

<!-- ^(common.topper) -->
<div class="row">
<div class="col-12">

<!-- ^(common.topper.table) -->
<table class="small table topper">

<!-- ^(common.topper.body) -->
<tbody>
<tr>
<td scope="row" class="border-0 font-weight-bold px-0 text-secondary">News Log</td>
</tr>
</tbody>
<!-- $(common.topper.body) -->

</table>
<!-- $(common.topper.table) -->

</div>
</div>
<!-- $(common.topper) -->

<!-- ^(news.row) -->
<div class="row">

<!-- ^(news.column) -->
<div class="col-12">

<!-- ^(common.table) -->
<table class="small table border mb-3">

<!-- ^(common.table.head) -->
<thead class="thead-light">
<tr>
<td scope="col" class="font-weight-bold text-dark">Monday, July 4th, 2022</td>
</tr>
</thead>
<!-- $(common.table.head) -->

<!-- ^(common.table.body) -->
<tbody>
<tr>
<td scope="row" class="position-relative"><img src="https://brunnerj.com/fairylab/images/teams/whitesox/whitesox-icon.png" width="16" height="16" border="0" class="absolute-icon left"><span class="d-block pl-4">1B <a href="https://statsplus.net/oblootp/player/33892">Viosergy Rosa</a> wins the MLB AL Player of the Week Award.</span></td>
</tr>
<tr>
<td scope="row" class="position-relative"><img src="https://brunnerj.com/fairylab/images/teams/braves/braves-icon.png" width="16" height="16" border="0" class="absolute-icon left"><span class="d-block pl-4">1B <a href="https://statsplus.net/oblootp/player/27194">Freddie Freeman</a> wins the MLB NL Player of the Week Award.</span></td>
</tr>
</tbody>
<!-- $(common.table.body) -->

</table>
<!-- $(common.table) -->

</div>
<!-- $(news.column) -->

</div>
<!-- $(news.row) -->

</div>
<!-- $(news.news) -->

<!-- ^(news.transactions) -->
<div class="container">

<!-- ^(common.topper) -->
<div class="row">
<div class="col-12">

<!-- ^(common.topper.table) -->
<table class="small table topper">

<!-- ^(common.topper.body) -->
<tbody>
<tr>
<td scope="row" class="border-0 font-weight-bold px-0 text-secondary">Major Transactions Log</td>
</tr>
</tbody>
<!-- $(common.topper.body) -->

</table>
<!-- $(common.topper.table) -->

</div>
</div>
<!-- $(common.topper) -->

<!-- ^(news.row) -->
<div class="row">

<!-- ^(news.column) -->
<div class="col-12">

<!-- ^(common.table) -->
<table class="small table border mb-3">

<!-- ^(common.table.head) -->
<thead class="thead-light">
<tr>
<td scope="col" class="font-weight-bold text-dark">Monday, July 4th, 2022</td>
</tr>
</thead>
<!-- $(common.table.head) -->

<!-- ^(common.table.body) -->
<tbody>
<tr>
<td scope="row" class="position-relative"><img src="https://brunnerj.com/fairylab/images/teams/padres/padres-icon.png" width="16" height="16" border="0" class="absolute-icon left"><span class="d-block pl-4">Signed <a href="https://statsplus.net/oblootp/player/56862">Antonio Marquis</a> to a minor league contract with a signing bonus of $180,000.</span></td>
</tr>
<tr>
<td scope="row" class="position-relative"><img src="https://brunnerj.com/fairylab/images/teams/obl/obl-icon.png" width="16" height="16" border="0" class="absolute-icon left"><span class="d-block pl-4">The Detroit Tigers traded 25-year-old minor league left fielder <a href="https://statsplus.net/oblootp/player/28672">Nick Plummer</a> to the Milwaukee Brewers, getting 33-year-old closer <a href="https://statsplus.net/oblootp/player/36352">Phil Klein</a> in return.</span></td>
</tr>
<tr>
<td scope="row" class="position-relative"><img src="https://brunnerj.com/fairylab/images/teams/obl/obl-icon.png" width="16" height="16" border="0" class="absolute-icon left"><span class="d-block pl-4">The Detroit Tigers traded 25-year-old minor league left fielder <a href="https://statsplus.net/oblootp/player/25202">Wesley Rodriguez</a> and 31-year-old left fielder <a href="https://statsplus.net/oblootp/player/33588">Michael A. Taylor</a> to the Chicago Cubs, getting 30-year-old right fielder <a href="https://statsplus.net/oblootp/player/35161">Drew Vettleson</a> in return.</span></td>
</tr>
<tr>
<td scope="row" class="position-relative"><img src="https://brunnerj.com/fairylab/images/teams/obl/obl-icon.png" width="16" height="16" border="0" class="absolute-icon left"><span class="d-block pl-4">The Detroit Tigers traded 21-year-old minor league shortstop <a href="https://statsplus.net/oblootp/player/32327">Jesus Martinez</a> to the Philadelphia Phillies, getting 28-year-old left fielder <a href="https://statsplus.net/oblootp/player/37237">Mitch Nay</a> in return.</span></td>
</tr>
<tr>
<td scope="row" class="position-relative"><img src="https://brunnerj.com/fairylab/images/teams/obl/obl-icon.png" width="16" height="16" border="0" class="absolute-icon left"><span class="d-block pl-4">The Minnesota Twins traded 25-year-old first baseman <a href="https://statsplus.net/oblootp/player/39581">Lewin Diaz</a> to the Chicago Cubs, getting 30-year-old starting pitcher <a href="https://statsplus.net/oblootp/player/37975">Matt Anderson</a>, 19-year-old minor league starting pitcher <a href="https://statsplus.net/oblootp/player/22921">Juan Monreal</a> and 21-year-old minor league catcher <a href="https://statsplus.net/oblootp/player/55093">Jonathan Salazar</a> in return.</span></td>
</tr>
</tbody>
<!-- $(common.table.body) -->

</table>
<!-- $(common.table) -->

</div>
<!-- $(news.column) -->

</div>
<!-- $(news.row) -->

</div>
<!-- $(news.transactions) -->

<!-- ^(common.dialog) -->
<div class="modal fade" id="menu" tabindex="-1" role="dialog"
     aria-labelledby="menu-label" aria-hidden="true">
<div class="modal-dialog" role="document">
<div class="modal-content">

<!-- ^(common.dialog.header) -->
<div class="modal-header d-block py-0">
<div class="head position-relative small text-secondary">
<img src="https://brunnerj.com/fairylab/favicon-32x32.png" width="16" height="16" border="0" class="absolute-icon left"><span class="d-block px-4">Fairylab</span>
<button type="button" class="close absolute-icon right m-0 p-0"
        data-dismiss="modal" aria-label="Close">
<span aria-hidden="true">&times;</span>
</button>
</div>
</div>
<!-- $(common.dialog.header) -->

<!-- ^(common.dialog.body) -->
<div class="modal-body pb-0">
<!-- ^(common.table) -->
<table class="small table topper">

<!-- ^(common.table.body) -->
<tbody>
<tr>
<td scope="row" class="border-0 font-weight-bold px-0 text-secondary">Site Links</td>
</tr>
</tbody>
<!-- $(common.table.body) -->

</table>
<!-- $(common.table) -->

<!-- ^(common.table) -->
<table class="small table border mb-3">

<!-- ^(common.table.head) -->
<thead class="thead-light">
<tr>
<td scope="col" class="font-weight-bold text-dark">Tasks</td>
</tr>
</thead>
<!-- $(common.table.head) -->

<!-- ^(common.table.body) -->
<tbody>
<tr>
<td scope="row" class="position-relative"><span class="oi oi-timer absolute-icon left text-secondary"></span><span class="d-block pl-4"><a href="/fairylab/gameday/">Gameday</a></span></td>
</tr>
<tr>
<td scope="row" class="position-relative"><span class="oi oi-spreadsheet absolute-icon left text-secondary"></span><span class="d-block pl-4"><a href="/fairylab/standings/">Standings</a></span></td>
</tr>
</tbody>
<!-- $(common.table.body) -->

</table>
<!-- $(common.table) -->

<!-- ^(common.table) -->
<table class="small table border mb-3">

<!-- ^(common.table.head) -->
<thead class="thead-light">
<tr>
<td scope="col" class="font-weight-bold text-dark">Other</td>
</tr>
</thead>
<!-- $(common.table.head) -->

<!-- ^(common.table.body) -->
<tbody>
<tr>
<td scope="row" class="position-relative"><span class="oi oi-dashboard absolute-icon left text-secondary"></span><span class="d-block pl-4"><a href="/fairylab/dashboard/">Dashboard</a></span></td>
</tr>
<tr>
<td scope="row" class="position-relative"><span class="oi oi-home absolute-icon left text-secondary"></span><span class="d-block pl-4"><a href="/fairylab/">Home</a></span></td>
</tr>
<tr>
<td scope="row" class="position-relative"><span class="oi oi-lightbulb absolute-icon left text-secondary"></span><span class="d-block pl-4"><a href="/fairylab/sandbox/">Sandbox</a></span></td>
</tr>
</tbody>
<!-- $(common.table.body) -->

</table>
<!-- $(common.table) -->

</div>
<!-- $(common.dialog.body) -->

</div>
</div>
</div>
<!-- $(common.dialog) --></main>
<!-- $(base.main) -->

<script src="https://code.jquery.com/jquery-3.2.1.slim.min.js"
        integrity="sha384-KJ3o2DKtIkvYIK3UENzmM7KCkRr/rE9/Qpg6aAZGJwFDMVNA/GpGFF93hXpG5KkN"
        crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/popper.js/1.12.9/umd/popper.min.js"
        integrity="sha384-ApNbgh9B+Y1QKtv3Rn7W3mgPxhU9K/ScQsAP7hUibX39j7fakFPskvXusvfa0b4Q"
        crossorigin="anonymous"></script>
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/4.0.0-beta.3/js/bootstrap.min.js"
        integrity="sha384-a5N7Y/aK3qNeh15eJKGWxsqtnX/wWdSZSKp+81YjTmS15nvnvxKHuzaWwXHDli+4"
        crossorigin="anonymous"></script>


</body>

</html>